### PR TITLE
Decomp scrcmd 5/7/8/12/19

### PR DESCRIPTION
**Pull request by Admiral-Fish**

### CODE REVIEW

File: include/unk_020518D8.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#ifndef POKEHEARTGOLD_UNK_020518D8_H
+#define POKEHEARTGOLD_UNK_020518D8_H
+
+BOOL IsBattleResultWin(u32 a0);
+int sub_02052564(u32 a0);
+int sub_02052574(u32 a0);
+
+#endif //POKEHEARTGOLD_UNK_020518D8_H
```
Comment (tgsm) [inline include/unk_020518D8.h:8]: Needs a newline

File: include/unk_020557E0.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+void sub_020557E0(FieldSystem *fsys);
+void sub_020557F8(FieldSystem *fsys);
+void sub_02055828(FieldSystem *fsys);
+void sub_0202EC70(struct UnkStruct_0202EB30 *a0, int i, POKEMON *pokemon);
+int sub_020558D0(FieldSystem *fsys);
+int sub_020558E0(FieldSystem *fsys);
+int sub_020558F0(FieldSystem *fsys);
+
+#endif //POKEHEARTGOLD_UNK_020557E0_H
```
Comment (tgsm) [inline include/unk_020557E0.h:15]: ditto

File: include/unk_020632B0.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef POKEHEARTGOLD_UNK_020632B0_H
+#define POKEHEARTGOLD_UNK_020632B0_H
+
+#include "script.h"
+
+int sub_0206457C(FieldSystem *fsys, LocalMapObject *localMapObject, FIELD_PLAYER_AVATAR *playerAvatar, int a3, int a4, int a5, int a6, u16 trainerNum);
+BOOL sub_02064598(void);
+void sub_020645AC(int a0);
+
+#endif //POKEHEARTGOLD_UNK_020632B0_H
```
Comment (tgsm) [inline include/unk_020632B0.h:10]: ditto

File: src/scrcmd_8.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    case 3: {
+        u16 groupId = ScriptGetVar(ctx);
+        u16 fieldNo = ScriptGetVar(ctx);
+        BufferGroupName(*msg, save, groupId, fieldNo, 1);
+        break;
+    }
+    case 4: {
+        u16 *r5 = sub_0202C7E0(group, 0, 0);
+        u16 *retPtr = ScriptGetVarPointer(ctx);
+        CreateNamingScreen(ctx->taskman, 5, 0, 7, 0, r5, retPtr);
```
Comment (tgsm) [inline src/scrcmd_8.c:49]:
```suggestion
        CreateNamingScreen(ctx->taskman, 5, 0, OT_NAME_LENGTH, 0, r5, retPtr);
```
As for the `5`, we should probably turn that into a `NAMINGSCREEN_UNK5` constant for now

File: src/scrcmd_8.c [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        if (count >= 4) {
+            *retPtr = TRUE;
+        } else {
+            *retPtr = FALSE;
+        }
+    }
+    }
+
+    return FALSE;
+}
```
Comment (tgsm) [inline src/scrcmd_8.c:88]: This file also needs a newline at the end

File: src/scrcmd_battle.c [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+BOOL ScrCmd_223(SCRIPTCONTEXT *ctx) {    
+    u32 *scriptUnk0C = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_0C);
+    SetupAndStartTrainerBattle(ctx->taskman, 1, 0, 0, 0, 0, 11, scriptUnk0C);
+    return TRUE;
+}
+
+BOOL ScrCmd_224(SCRIPTCONTEXT *ctx) {
+    LocalMapObject **lastTalked = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_LAST_TALKED);
+    u32 mapId = MapObject_GetID(*lastTalked);
```
Comment (tgsm) [inline src/scrcmd_battle.c:247]:
```suggestion
    u32 mapObjectId = MapObject_GetID(*lastTalked);
```

File: src/scrcmd_battle.c [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    TrainerFlagSet(ctx->fsys->savedata, (u16)mapId);
+    return FALSE;
+}
+
+BOOL ScrCmd_225(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    LocalMapObject **lastTalked = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_LAST_TALKED);
+    u32 var0 = ScriptReadWord(ctx);
+    u32 mapId = MapObject_GetID(*lastTalked);
```
Comment (tgsm) [inline src/scrcmd_battle.c:256]:
```suggestion
    u32 mapObjectId = MapObject_GetID(*lastTalked);
```

File: src/scrcmd_battle.c [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    LocalMapObject **lastTalked = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_LAST_TALKED);
+    u32 mapId = MapObject_GetID(*lastTalked);
+    TrainerFlagSet(ctx->fsys->savedata, (u16)mapId);
+    return FALSE;
+}
+
+BOOL ScrCmd_225(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    LocalMapObject **lastTalked = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_LAST_TALKED);
+    u32 var0 = ScriptReadWord(ctx);
```
Comment (tgsm) [inline src/scrcmd_battle.c:255]: `offset`?

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/scrcmd_8.c (hunk 4 above)
Comment (PikalaxALT) [reply src/scrcmd_8.c:49]: Maybe `NAMINGSCREEN_GROUP`?

File: src/scrcmd_battle.c [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+BOOL ScrCmd_224(SCRIPTCONTEXT *ctx) {
+    LocalMapObject **lastTalked = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_LAST_TALKED);
+    u32 mapId = MapObject_GetID(*lastTalked);
+    TrainerFlagSet(ctx->fsys->savedata, (u16)mapId);
+    return FALSE;
+}
+
+BOOL ScrCmd_225(SCRIPTCONTEXT *ctx) {
```
Comment (PikalaxALT) [inline src/scrcmd_battle.c:252]:
```suggestion
BOOL ScrCmd_GotoIfTrainerDefeated(SCRIPTCONTEXT *ctx) {
```

**Review by PikalaxALT — APPROVED** [review]

**Review by tgsm — APPROVED** [review]

**Pull request merged**